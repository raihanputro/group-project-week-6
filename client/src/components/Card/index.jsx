import classes from './style.module.scss';
import { useNavigate } from 'react-router-dom';
import { connect } from 'react-redux';

import { createStructuredSelector } from 'reselect';

const Card = ({ data  }) => {
  const navigate = useNavigate();
  const gotToDetail = () => {
    navigate(`/member/task/${data?.task_id}`);
  };

  return (
    <div className={classes.container} onClick={gotToDetail}>
      <div className={classes.content}>
        <p className={classes.title}>{data?.Task?.name} ({data?.Task?.User?.name})</p>
        <p className={classes.date}>{data?.Task?.start_date.substring(0,10)} s/d {data?.Task?.end_date.substring(0,10)}</p>
        <p className={classes.shortdesc}>{data?.Task?.description}</p>
      </div>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({});

export default connect(mapStateToProps)(Card);