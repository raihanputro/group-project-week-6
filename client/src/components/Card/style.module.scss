@import '../../styles/base.scss';

.container {
  @include directionY;
  width: 300px;
  cursor: pointer;
  height: 362px;
  position: relative;
  border-radius: 5px;
  background: #fff;
  box-shadow: 0px 0px 15px 0px rgba(82, 82, 82, 0.18);
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
  .image {
    width: 100%;
    height: 180px;
    border-radius: 5px;
    -webkit-border-radius: 5px;
    -moz-border-radius: 5px;
    -ms-border-radius: 5px;
    -o-border-radius: 5px;
  }
  .containerCircle {
    position: absolute;
    top: 13px;
    right: 13px;
    .circleBook {
      cursor: pointer;
      display: flex;
      justify-content: center;
      align-items: center;
      width: 30px;
      height: 30px;
      background-color: white;
      border-radius: 30px;
      -webkit-border-radius: 30px;
      -moz-border-radius: 30px;
      -ms-border-radius: 30px;
      -o-border-radius: 30px;
      .bookmark {
        width: 20px;
        height: 20px;
      }
    }
    .circleBookActive{
      @extend .circleBook;
      background-color: blue;
    }
  }
  .content {
    @include directionY;
    margin: 22px 15px 0 15px;
    .title {
      color: black;
      font-family: 'Poppins', sans-serif;
      font-size: 14px;
      font-style: normal;
      font-weight: 700;
      line-height: normal;
    }
    .date {
      color: grey;
      font-family: 'Poppins', sans-serif;
      font-size: 12px;
      font-style: normal;
      font-weight: 400;
      line-height: normal;
      margin-bottom: 16px;
    }
    .shortdesc {
      color: grey;
      font-family: 'Poppins', sans-serif;
      font-size: 12px;
      font-style: normal;
      font-weight: 400;
      line-height: normal;
    }
  }
}