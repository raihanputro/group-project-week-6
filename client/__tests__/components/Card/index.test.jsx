import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import RouterDom from 'react-router-dom';

import Card from '@components/Card';
import store from '../../../src/configureStore';

let wrapper;

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
}));

jest.mock('reselect', () => ({
    ...jest.requireActual('reselect'),
}));

const ParentComponent = (children) => (
    <Provider store={store}>
        {children}
    </Provider>
);


const mockProps = {
    Task: {
        name: 'Bowo',
        User: {
            name: 'Tono'
        },
        start_date: '15-02-2024',
        end_date: '18-02-2024',
        description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Asperiores odit soluta laboriosam nisi repellat eum maiores blanditiis dicta facere hic maxime, est et aliquam, dolorem, consequuntur nobis adipisci consequatur sequi?'
    }
};


describe('Card Test', () => {
    beforeEach(() => {
        wrapper = render(ParentComponent(<Card data={mockProps} />));
    });

    test('should render correctly', () => {
        const { getByTestId } = wrapper;
        expect(getByTestId('card')).toBeInTheDocument();
    });

    test('Card clicked', () => {
        const { getByTestId } = wrapper;
        expect(getByTestId('card')).toBeInTheDocument();
        fireEvent.click(getByTestId('card'))
    });

    test('Should match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});