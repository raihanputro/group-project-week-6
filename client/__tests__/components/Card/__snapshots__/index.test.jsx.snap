// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="card"
      >
        <div
          class="content"
          data-testid="card-content"
        >
          <p
            class="title"
          >
            Bowo
             (
            Tono
            )
          </p>
          <p
            class="date"
          >
            15-02-2024
             s/d 
            18-02-2024
          </p>
          <p
            class="shortdesc"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Asperiores odit soluta laboriosam nisi repellat eum maiores blanditiis dicta facere hic maxime, est et aliquam, dolorem, consequuntur nobis adipisci consequatur sequi?
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="card"
    >
      <div
        class="content"
        data-testid="card-content"
      >
        <p
          class="title"
        >
          Bowo
           (
          Tono
          )
        </p>
        <p
          class="date"
        >
          15-02-2024
           s/d 
          18-02-2024
        </p>
        <p
          class="shortdesc"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Asperiores odit soluta laboriosam nisi repellat eum maiores blanditiis dicta facere hic maxime, est et aliquam, dolorem, consequuntur nobis adipisci consequatur sequi?
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
